README.md

```diff
@@ -48,9 +48,43 @@ A quick note on environment variables
 
 Use `.env.local` for local development.
 
+### Server Side Environment Variables
+
+If a variable is required ONLY on the SERVER side (ie. in an api endpoint), then the variable can be declared in `.env.local.`
+
+SERVER side methods in Next.js read environments at RUN TIME, which means they have access to environment variables on the server - these can be found by running `env` or referenced within the app as `process.env.ENV`. These variables can/should be managed in QA and Production environments by updating the Task Definition and/or Cloud Formation Templates.
+
 SERVER side references of environment variables can use `process.env.ENV`
 
-If you are referencing an environment variable on the CLIENT, the environment variable needs to be added to `nextConfig.env` in `next.config.js.` Please use the process.env.ENV reference for the variable.
+### Client Side Environment Variables
+
+If a variable is required ONLY on the CLIENT, the environment variable needs to be declared in `appConfig.ts`.
+
+CLIENT side methods in Next.js only have access to environment variables that are declared at BUILD TIME, which means they DO NOT have access to environment variables on the server and thus do not have access to `process.env.ENV`. These variables should be managed in QA and Production environments by either hard coding them in `appConfig.ts` AND/OR declaring them in the Dockerfile and travis.yml.
```

Comment:
This is partly true.

Another way to declare client-side variables is by prefixing them with `NEXT_PUBLIC_`. Then they _are_ accessible in the app as `process.env.NEXT_PUBLIC_[VAR_NAME]`.

But, they **must** be declared at build time like you mentioned.

This pattern of using `APP_ENV` and the hard coded values in `appConfig.ts` works well but you _could_ set client side env vars through docker/travis. For example, you can set `NEXT_PUBLIC_ADOBE_URL="prod-url"` in travis.yml and send it to the dockerfile so it's there at build time for the app. Then in the app, you can use `process.env.NEXT_PUBLIC_ADOBE_URL` on both the client and server.

You don't have to mention this if the decided pattern is to use `appConfig.ts` and the `APP_ENV` variables. This convention/pattern works well and I'm happy to support this, but we should choose one and we should be clear on the alternatives in case questions come up.